### PR TITLE
Search disapearing fix

### PR DESCRIPTION
**Pull request by thecalcc**
Fix CPCN-50
## Checklist

- [x] This pull request is not adding new forms that use redux
- [x] This pull request is adding missing TypeScript types to modified code segments where it's easy to do so with confidence
- [x] This pull request is replacing `lodash.get` with optional chaining for modified code segments

### CODE REVIEW

**Comment by thecalcc** [comment]
Currently there's a bug, when you type something the buckets are getting filtered by search term and in the case where there are > 50 buckets we hide the searchbar which isn't what we want.

File: assets/wire/components/filters/FilterGroup.tsx [hunk 1]
```diff
@@ -70,16 +70,17 @@ export default function FilterGroup({group, activeFilter, aggregations, toggleFi
...
-    )
+    );
+    const bucketsBySearchTerm = allBuckets
         .sort(compareFunction)
         .filter(({key}: any) => searchTerm.length > 0 ? key.toString().toLocaleLowerCase().includes(searchTerm) : true);
 
     return (
         <NavGroup key={group.field} label={group.label}>
-            {buckets.length > 50 && (
+            {allBuckets.length > 50 && (
```
Comment (tomaskikutis) [inline assets/wire/components/filters/FilterGroup.tsx:83]: why `50`? shouldn't you use a `pageSize` variable here?

Comment (thecalcc) [reply assets/wire/components/filters/FilterGroup.tsx:83]: 50 was in the ticket requirement

File: assets/wire/components/filters/FilterGroup.tsx [hunk 2]
```diff
@@ -8,6 +8,7 @@ import FilterItem from './FilterItem';
 import {WithPagination} from 'components/pagination/WithPagination';
 import {searchIcon} from 'search/components/search-icon';
 
+const SEARCHBAR_THRESHOLD = 50;
```
Comment (tomaskikutis) [inline assets/wire/components/filters/FilterGroup.tsx:11]: can you add a comment on what exactly was the requirement?

**Review by tomaskikutis — APPROVED** [review]

**Pull request merged**